src/solve.jl

```diff
@@ -50,7 +50,15 @@ To see all available input options, see
 function solve(system::InitialValueProblem,
                options::Options;
                op::ContinuousPost=default_operator(system))
-    solve!(system, copy(options), op=op)
+    res = solve!(system, copy(options), op=op)
+    merged_options = merge(res.options.defaults, res.options.specified)
+    if res isa ReachSolution
```

Comment:
Can we use `if merged_options[:mode] == ...` instead?

Reply:
Or even `options[:mode]`, since this isn't supposed to be touched by the solver.

Reply:
What?